data_management_system.rb

```diff
@@ -0,0 +1,87 @@
+# require 'pry'
+
+class Writer
+  def booking(book, _path, file)
+    data = {
+      title: book.title,
+      author: book.author
+    }
+    file.puts(data.to_json)
+    file.close
+  end
+
+  def book_writer(books, path)
+    file = if File.exist?(path)
+             File.open(path, 'a')
+           else
+             File.open(path, 'w')
+           end
+    books.each do |book|
+      booking(book, path, file) if book.is_a?(Book)
+    end
+  end
+
+  def student(person, _path, file)
+    data = {
+      id: person.id,
+      age: person.age,
+      name: person.name,
+      parent_permission: person.instance_variable_get(:@parent_permission)
+    }
+    file.puts(data.to_json)
+    file.close
+  end
+
+  def teacher(person, _path, file)
+    data = {
+      id: person.id,
+      age: person.age,
+      name: person.name,
+      specialization: person.instance_variable_get(:@specialization),
+      parent_permission: person.instance_variable_get(:@parent_permission)
+    }
+    file.puts(data.to_json)
+    file.close
+  end
+
+  def people_writer(peoples, path)
+    file = if File.exist?(path)
+             File.open(path, 'a')
+           else
+             File.open(path, 'w')
+           end
+    peoples.each do |people|
+      student(people, path, file) if people.is_a?(Student)
+      teacher(people, path, file) if people.is_a?(Teacher)
+    end
+  end
```

Comment:
- [ ] After I try to create a person and then close the app with 7 I get the following error, kindly check into that. Also the name on the json file saves differently from what I input 
![image](https://github.com/Lornakaboro/School-library/assets/62705947/7d6496a5-e0e7-4fdc-9190-a00d50c3927c)
![image](https://github.com/Lornakaboro/School-library/assets/62705947/d38096b0-e631-4215-b35e-b347d1d40d71)
- [ ] It saves always the same name as the first one 
![image](https://github.com/Lornakaboro/School-library/assets/62705947/733d32fe-bf83-4fab-937f-6912c9a1b59c)

Reply:
- [ ] Even if I just run the app and exit it with 7 automatically one person gets saved, check into this as well please